fondant/component.py

```diff
@@ -264,27 +264,32 @@ def transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
             dataframe: A Pandas dataframe containing a partition of the data
         """
 
-    def wrapped_transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
-        """Method wrapping the transform method to switch between hierarchical and flattened
-        columns.
-        """
-        dataframe.columns = pd.MultiIndex.from_tuples(
-            tuple(column.split("_")) for column in dataframe.columns
-        )
-        dataframe = self.transform(dataframe)
-        # Drop columns not in the produces section of the component spec
-        dataframe = dataframe.drop(
-            columns=[
-                (subset, field)
-                for (subset, field) in dataframe.columns
-                if subset not in self.spec.produces
-                or field not in self.spec.produces[subset].fields
-            ],
-        )
-        dataframe.columns = [
-            "_".join(column) for column in dataframe.columns.to_flat_index()
-        ]
-        return dataframe
+    @staticmethod
+    def wrap_transform(transform: t.Callable, *, spec: ComponentSpec) -> t.Callable:
+        def wrapped_transform(dataframe: pd.DataFrame) -> pd.DataFrame:
```

Comment:
Can you re-add the deleted docstring 

  """Method wrapping the transform method to switch between hierarchical and flattened
  columns.
  """